crates/sui-authority-aggregation/src/lib.rs

```diff
@@ -54,9 +66,18 @@ where
     FMap: FnOnce(K, Arc<Client>) -> AsyncResult<'a, V, E> + Clone + 'a,
     FReduce: Fn(S, K, StakeUnit, Result<V, E>) -> BoxFuture<'a, ReduceOutput<R, S>>,
 {
-    let authorities_shuffled = committee.shuffle_by_stake(authority_preferences, None);
+    let (preference, prefetch_timeout) = if let Some(SigRequestPrefs {
+        ordering_pref,
+        prefetch_timeout,
+    }) = authority_preferences
+    {
+        (Some(ordering_pref), Some(prefetch_timeout))
+    } else {
+        (None, None)
+    };
+    let authorities_shuffled = committee.shuffle_by_stake(preference.as_ref(), None);
```

Comment:
honestly I think we can remove the `preference` because it's hacky and more importantly it doesn't really work. But let's leave this for a future PR